### ISSUE TITLE
Small object allocation and shot detection filter

### ISSUE THREAD

**Issue by phrack** [phrack/ShootOFF#399]
It is widely supported that if you have code that has high CPU utilization that needs to be fast you should not do object allocation unless 100% necessary. This is both because allocating objects is slow and because allocating a large number of small objects ensures the GC will get in the way. 

In the filter update code, which currently consumes 42% of an average ShootOFF executions CPU time, there are several small object allocations we can probably get rid of:

https://github.com/phrack/ShootOFF/blob/master/src/main/java/com/shootoff/camera/shotdetection/ShotDetectionManager.java
https://github.com/phrack/ShootOFF/blob/master/src/main/java/com/shootoff/camera/shotdetection/ShotDetectionManager.java#L366

The HSV byte array is an object under the hood even though the type is a primitive. Making this an integer and doing bit shifts to get the HSV values is almost definitely faster. Even better: just pass them as separate parameters to avoid the need to create an object AND the need to bit shift. Same thing with the maskInt. We're making thousands of small objects for every frame where a primitive could do. Optionals? Super handy and good for clean code! In this very narrow case? Tons of small object allocations where we could just do a null check. Pixel? Well, I dunno, but there might be a better way.

In short, these ~40 lines of code are so performance critical that it's OK for them to be a bit ugly if it avoids an object allocation.
